### DISCUSSION TITLE
Test behaves differently depending on how it's run

### DISCUSSION THREAD

**Discussion by 1Mark** [pytest-dev/pytest#10233] [Q&A]
Is there any difference in the setup done when running a test using the file path Vs running it against a directory Vs running it using the pytest -m someMarker.

I have test that only passes when the whole directory runs but not when I run it individually. 

The test is as follows
```
def test_var():
    assert semaphore.count == 0
    doSomething()
    assert semaphore.count == 1 # this fails when running the directory 
```
I'm struggling to understand what could cause this behaviour.

**Comment by nicoddemus** [comment]
The usual cause is that some other test is changing global state that is affecting this test.

You can use [detect-test-pollution](https://github.com/asottile/detect-test-pollution) to nail down which test is affecting `test_var`.

**Reply by 1Mark** [reply]
Hmm agreed but the first assertion passes constantly but the second fails. Which means the only thing that could affect it is some setup or something since the only thing that could impact it is something associated to `doSomething()`

I wasn't aware of the package you mentioned so will give it a shot.
